### PR TITLE
made changes to be compliant w scapy sockets

### PR DESCRIPTION
**Pull request by jfeiglin**

### CODE REVIEW

File: mitm_service/protocol_filters.py [hunk 1]
```diff
@@ -22,34 +22,24 @@ def generate_ip_redirect_rule(target_ip, redirect_ip):
     target_ip = inet_aton(target_ip)
     redirect_ip = inet_aton(redirect_ip)
 
-    def disrupt_ip_traffic(ip_header, ip_payload):
-        if not isinstance(ip_header, IpHeader):
+    def disrupt_ip_traffic(scapy_pkt):
```
Comment (benjaminbaker-dev) [inline mitm_service/protocol_filters.py:25]: change docstr so that input and return are scapy packets :)

File: mitm_service/protocol_filters.py [hunk 2]
```diff
@@ -22,34 +22,24 @@ def generate_ip_redirect_rule(target_ip, redirect_ip):
...
+        if IP not in scapy_pkt:
             # not an ip packet so ignore
-            return ip_header, ip_payload
+            return scapy_pkt
 
-        if ip_header.dst_ip != target_ip and ip_header.src_ip != redirect_ip:
+        ip_header = scapy_pkt[IP]
+        if ip_header.dst != target_ip and ip_header.src != redirect_ip:
             return ip_header, ip_payload
```
Comment (benjaminbaker-dev) [inline mitm_service/protocol_filters.py]: should be return scapy_pakt

File: mitm_service/protocol_filters.py [hunk 3]
```diff
@@ -22,34 +22,24 @@ def generate_ip_redirect_rule(target_ip, redirect_ip):
...
-            l4_header, l4_header_len = UdpHeader.parse_raw_header(ip_payload)
-        else:
-            return ip_header, ip_payload
-        l4_payload = ip_payload[l4_header_len:]
-        l4_header.checksum = 0
-        ip_payload = l4_header.get_raw_header() + l4_payload
+            ip_header.src = target_ip
 
-        return ip_header, ip_payload
+        return scapy_pkt
```
Comment (benjaminbaker-dev) [inline mitm_service/protocol_filters.py:42]: although changes will persist w/o, i think explicit set of scapy_pakt[IP] = ip_header before return of scapy_pakt makes it much more readable

**Review by benjaminbaker-dev — CHANGES_REQUESTED** [review]
:)

File: mitm_service/protocol_filters.py (hunk 3 above)
Comment (jfeiglin) [reply mitm_service/protocol_filters.py:42]: yeah thats just wrong no

Comment (benjaminbaker-dev) [reply mitm_service/protocol_filters.py:42]: lol

**Review by benjaminbaker-dev — APPROVED** [review]

**Pull request merged**